explorer/interpreter/interpreter.cpp

```diff
@@ -1534,17 +1530,18 @@ auto Interpreter::StepExp() -> ErrorOr<Success> {
         // -> { {C',E',F'} :: {C, E, F} :: S, H}
         // Prepare parameters tuple.
         std::vector<Nonnull<const Value*>> param_values;
-        for (int i = 1; i <= num_args; ++i) {
-          param_values.push_back(act.results()[i]);
+        for (const auto& arg_result :
+             llvm::ArrayRef(act.results()).drop_front().take_front(num_args)) {
+          param_values.push_back(arg_result);
         }
         const auto* param_tuple = arena_->New<TupleValue>(param_values);
         // Prepare witnesses.
         ImplWitnessMap witnesses;
         if (num_witnesses > 0) {
-          int i = 1 + num_args;
-          for (const auto& [impl_bind, impl_exp] : call.witnesses()) {
-            witnesses[impl_bind] = act.results()[i];
-            ++i;
+          for (const auto& [witness, result] : llvm::zip(
+                   call.witnesses(),
+                   llvm::ArrayRef(act.results()).drop_front(1 + num_args))) {
+            witnesses[witness.first] = result;
           }
         }
```

Comment:
Maybe we should consider adding a `Split` utility so we can do this:

```suggestion
        std::vector<Nonnull<const Value*>> param_values;
        auto [arg_results, witness_results] =
            Split(llvm::ArrayRef(act.results()).drop_front(), num_args);
        for (const auto& arg_result : arg_results) {
          param_values.push_back(arg_result);
        }
        const auto* param_tuple = arena_->New<TupleValue>(param_values);
        // Prepare witnesses.
        ImplWitnessMap witnesses;
        if (num_witnesses > 0) {
          for (const auto& [witness, result] : llvm::zip(call.witnesses(), witness_results)) {
            witnesses[witness.first] = result;
          }
        }
```

(No action expected in this PR though.)

Reply:
Acknowledged, although it seems not as worth it if we don't get more use-cases... (though, I can use `slice`, might go and fix that)